### PR TITLE
fix(components): [select-v2] label display error when re-rendering

### LINKED ISSUE #18035: [Component] [select-v2] ElSelectV2 - Selected option(s) changes from Label to Value when option is reset via @blur
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.38`
- Element Plus Version: `2.8.1`
- Browser / OS: `Chrome`
- Build Tool: `Vite`

### Related Component

- `el-select-v2`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtc2VsZWN0LXYyXG4gICAgdi1tb2RlbD1cInZhbHVlXCJcbiAgICBtdWx0aXBsZVxuICAgIGZpbHRlcmFibGVcbiAgICByZW1vdGVcbiAgICA6ZmlsdGVyLW1ldGhvZD1cInJlbW90ZU1ldGhvZFwiXG4gICAgY2xlYXJhYmxlXG4gICAgOm9wdGlvbnM9XCJvcHRpb25zXCJcbiAgICA6bG9hZGluZz1cImxvYWRpbmdcIlxuICAgIHBsYWNlaG9sZGVyPVwiUGxlYXNlIGVudGVyIGEga2V5d29yZFwiXG4gICAgY29sbGFwc2UtdGFnc1xuICAgIGNvbGxhcHNlLXRhZ3MtdG9vbHRpcFxuICAgIEBibHVyPVwib3B0aW9ucyA9IFtdXCJcbiAgLz5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5cbmNvbnN0IHN0YXRlcyA9IFtcbiAgJ0FsYWJhbWEnLFxuICAnQWxhc2thJyxcbiAgJ0FyaXpvbmEnLFxuICAnQXJrYW5zYXMnLFxuICAnQ2FsaWZvcm5pYScsXG4gICdDb2xvcmFkbycsXG4gICdDb25uZWN0aWN1dCcsXG4gICdEZWxhd2FyZScsXG4gICdGbG9yaWRhJyxcbiAgJ0dlb3JnaWEnLFxuICAnSGF3YWlpJyxcbiAgJ0lkYWhvJyxcbiAgJ0lsbGlub2lzJyxcbiAgJ0luZGlhbmEnLFxuICAnSW93YScsXG4gICdLYW5zYXMnLFxuICAnS2VudHVja3knLFxuICAnTG91aXNpYW5hJyxcbiAgJ01haW5lJyxcbiAgJ01hcnlsYW5kJyxcbiAgJ01hc3NhY2h1c2V0dHMnLFxuICAnTWljaGlnYW4nLFxuICAnTWlubmVzb3RhJyxcbiAgJ01pc3Npc3NpcHBpJyxcbiAgJ01pc3NvdXJpJyxcbiAgJ01vbnRhbmEnLFxuICAnTmVicmFza2EnLFxuICAnTmV2YWRhJyxcbiAgJ05ldyBIYW1wc2hpcmUnLFxuICAnTmV3IEplcnNleScsXG4gICdOZXcgTWV4aWNvJyxcbiAgJ05ldyBZb3JrJyxcbiAgJ05vcnRoIENhcm9saW5hJyxcbiAgJ05vcnRoIERha290YScsXG4gICdPaGlvJyxcbiAgJ09rbGFob21hJyxcbiAgJ09yZWdvbicsXG4gICdQZW5uc3lsdmFuaWEnLFxuICAnUmhvZGUgSXNsYW5kJyxcbiAgJ1NvdXRoIENhcm9saW5hJyxcbiAgJ1NvdXRoIERha290YScsXG4gICdUZW5uZXNzZWUnLFxuICAnVGV4YXMnLFxuICAnVXRhaCcsXG4gICdWZXJtb250JyxcbiAgJ1ZpcmdpbmlhJyxcbiAgJ1dhc2hpbmd0b24nLFxuICAnV2VzdCBWaXJnaW5pYScsXG4gICdXaXNjb25zaW4nLFxuICAnV3lvbWluZycsXG5dXG5jb25zdCBsaXN0ID0gc3RhdGVzLm1hcCgoaXRlbSk6IExpc3RJdGVtID0+IHtcbiAgcmV0dXJuIHsgdmFsdWU6IGB2YWx1ZToke2l0ZW19YCwgbGFiZWw6IGBsYWJlbDoke2l0ZW19YCB9XG59KVxuXG5pbnRlcmZhY2UgTGlzdEl0ZW0ge1xuICB2YWx1ZTogc3RyaW5nXG4gIGxhYmVsOiBzdHJpbmdcbn1cblxuY29uc3QgdmFsdWUgPSByZWYoW10pXG5jb25zdCBvcHRpb25zID0gcmVmPExpc3RJdGVtW10+KFtdKVxuY29uc3QgbG9hZGluZyA9IHJlZihmYWxzZSlcblxuY29uc3QgcmVtb3RlTWV0aG9kID0gKHF1ZXJ5OiBzdHJpbmcpID0+IHtcbiAgaWYgKHF1ZXJ5ICE9PSAnJykge1xuICAgIGxvYWRpbmcudmFsdWUgPSB0cnVlXG4gICAgc2V0VGltZW91dCgoKSA9PiB7XG4gICAgICBsb2FkaW5nLnZhbHVlID0gZmFsc2VcbiAgICAgIG9wdGlvbnMudmFsdWUgPSBsaXN0LmZpbHRlcigoaXRlbSkgPT4ge1xuICAgICAgICByZXR1cm4gaXRlbS5sYWJlbC50b0xvd2VyQ2FzZSgpLmluY2x1ZGVzKHF1ZXJ5LnRvTG93ZXJDYXNlKCkpXG4gICAgICB9KVxuICAgIH0sIDUwMClcbiAgfSBlbHNlIHtcbiAgICBvcHRpb25zLnZhbHVlID0gW11cbiAgfVxufVxuPC9zY3JpcHQ+XG4iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvZGlzdC9pbmRleC5jc3MnLCAnaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmZ1bGwubWluLm1qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzL1wiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9cIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6e319)

### Steps to reproduce

1. Use the ElSelectV2 Component
2. Add @blur event to component
3. Set @blur="options=[]"
4. Click in and out of the component field (to trigger blur event)
5. Notice of selected options change from Label to Value

### What is Expected?

Only expect label to show

### What is actually happening?

selected options show value as label

### Additional comments

Somewhat related to closed issue. https://github.com/element-plus/element-plus/issues/17343

I have not tested with other events. Either this is a blur + options reset issue or just pure reactive options fault
I have not tested select-v1

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #18035

### CODE REVIEW

File: packages/components/select-v2/src/useSelect.ts [hunk 1]
```diff
@@ -725,20 +725,31 @@ const useSelect = (props: ISelectV2Props, emit) => {
 
       return option
     }
+    let label = value
+    if (cachedOptions && cachedOptions.length) {
+      const option = cachedOptions.find(
+        (option) => getValueKey(getValue(option)) === selectValue
+      )
+      if (option) {
+        label = getLabel(option)
```
Comment (tolking) [inline packages/components/select-v2/src/useSelect.ts:734]:
```suggestion
        return option
```

Why not just return the `option` directly

Comment (btea) [reply packages/components/select-v2/src/useSelect.ts:734]: It seems to be OK, I fell into the misunderstanding of the previous code.

**Comment by prherranz** [comment]
Bump! Can be this issue solved for the next version?

**Review by tolking — APPROVED** [review]

**Review by makedopamine — APPROVED** [review]
Looks good :)

**Comment by btea** [comment]
> Bump! Can be this issue solved for the next version?

It has been merged and will be released in the next version.

**Comment by prherranz** [comment]
Thanks! Really appreciated!

**Pull request merged**